$NetBSD: patch-ac,v 1.4 2020/04/03 22:06:17 wiz Exp $

Fix various target paths.
Add a rule in order not to depend on gmake.
Honor CFLAGS and LDFLAGS.

--- Makefile.orig	2013-04-20 17:51:58.000000000 +0000
+++ Makefile
@@ -11,7 +11,7 @@
 # determination of the actual installation directories.
 # Suggested values are "/usr/local", "/usr", "/pkgs/fdupes-$(VERSION)"
 #
-PREFIX = /usr/local
+PREFIX ?= /usr/local
 
 #
 # When compiling for 32-bit systems, FILEOFFSET_64BIT must be enabled
@@ -50,13 +50,13 @@ PROGRAM_NAME=fdupes
 # BIN_DIR indicates directory where program is to be installed. 
 # Suggested value is "$(PREFIX)/bin"
 #
-BIN_DIR = $(PREFIX)/bin
+BIN_DIR = $(DESTDIR)$(PREFIX)/bin
 
 #
 # MAN_DIR indicates directory where the fdupes man page is to be 
 # installed. Suggested value is "$(PREFIX)/man/man1"
 #
-MAN_BASE_DIR = $(PREFIX)/man
+MAN_BASE_DIR = $(DESTDIR)$(PREFIX)/${PKGMANDIR}
 MAN_DIR = $(MAN_BASE_DIR)/man1
 MAN_EXT = 1
 
@@ -80,7 +80,7 @@ MKDIR   = mkdir -p
 CC = gcc
 COMPILER_OPTIONS = -Wall -O -g
 
-CFLAGS= $(COMPILER_OPTIONS) -I. -DVERSION=\"$(VERSION)\" $(EXTERNAL_MD5) $(OMIT_GETOPT_LONG) $(FILEOFFSET_64BIT)
+CFLAGS+= $(COMPILER_OPTIONS) -I. -DVERSION=\"$(VERSION)\" $(EXTERNAL_MD5) $(OMIT_GETOPT_LONG) $(FILEOFFSET_64BIT)
 
 INSTALL_PROGRAM = $(INSTALL) -c -m 0755
 INSTALL_DATA    = $(INSTALL) -c -m 0644
@@ -91,7 +91,9 @@ INSTALL_DATA    = $(INSTALL) -c -m 0644
 #
 #ADDITIONAL_OBJECTS = getopt.o
 
-OBJECT_FILES = fdupes.o md5/md5.o $(ADDITIONAL_OBJECTS)
+OBJECT_FILES = fdupes.o md5.o $(ADDITIONAL_OBJECTS)
+
+md5.o: md5/md5.c
 
 #####################################################################
 # no need to modify anything beyond this point                      #
@@ -100,7 +102,7 @@ OBJECT_FILES = fdupes.o md5/md5.o $(ADDI
 all: fdupes
 
 fdupes: $(OBJECT_FILES)
-	$(CC) $(CFLAGS) -o fdupes $(OBJECT_FILES)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o fdupes $(OBJECT_FILES)
 
 installdirs:
 	test -d $(BIN_DIR) || $(MKDIR) $(BIN_DIR)
